src/web/hooks/usePagination.js

```diff
@@ -0,0 +1,54 @@
+/* SPDX-FileCopyrightText: 2024 Greenbone AG
+ *
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import {useCallback} from 'react';
+
+/**
+ * Hook to to get the filter for the next, previous, last and first page for a
+ * list of entities
+ *
+ * @example
+ *
+ * const gmp = useGmp();
+ * const [filter, setFilter] = useState(new Filter());
+ * const [entities, setEntities] = useState([]);
+ * const [counts, setCounts] = useState({});
+ * const updateFilter = useCallback(filter => {
+ *   setFilter(filter);
+ *   gmp.tasks.get(filter).then(response => {setEntities(response.data);setCounts(response.meta.counts)});
+ * }, [gmp.tasks]);
+ * const [first, last, next, previous] = usePagination(filter, counts, updateFilter)
+ *
+ * @param {Filter} filter Current applied filter
+ * @param {Object} counts Current entities counts. Required for calculating the
+ *   last page.
+ * @param {Function} changeFilter Function to call when the new filter is applied
+ * @returns {Array} Tuple of functions to update the filter for the first, last,
+ *   next and previous page.
+ */
+const usePagination = (filter, counts, changeFilter) => {
+  const getNext = useCallback(() => {
+    changeFilter(filter.next());
```

Comment:
Question: Probably fine and safe, but  `filter` is always mutating?

Reply:
A new instance is created and returned for next, previous, last and first methods.